app/src/main/java/com/douglasbruce/tasky/MainActivity.kt

```diff
@@ -3,22 +3,52 @@ package com.douglasbruce.tasky
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.rememberNavController
 import com.douglasbruce.tasky.core.designsystem.theme.TaskyTheme
 import com.douglasbruce.tasky.navigation.TaskyNavHost
+import com.douglasbruce.tasky.MainActivityUiState.Loading
+import com.douglasbruce.tasky.MainActivityUiState.Success
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
+    val viewModel: MainActivityViewModel by viewModels()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
-        // TODO: Tie to uiState to keep splashScreen visible until we can determine if user is authenticated or not
+        var uiState: MainActivityUiState by mutableStateOf(Loading)
+
+        // Update the uiState
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.uiState
+                    .onEach {
+                        uiState = it
+                    }
+                    .collect()
+            }
+        }
+
         splashScreen.setKeepOnScreenCondition {
-            false
+            when (uiState) {
+                Loading -> true
+                is Success -> false
+            }
         }
```

Comment:
Couldn't you just refer to the ViewModel's `uiState` here? I don't see how the extra `uiState` initialized above and the lifecycle scope block is necessary at this point.

The `setKeepOnScreenCondition` should check the value of the state during each frame, so you shouldn't need your own variable in the activity here

Reply:
Will simplify to refer directly to the viewModels state and remove the extra initialization and lifecycle scope